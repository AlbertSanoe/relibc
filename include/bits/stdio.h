#ifndef _BITS_STDIO_H
#define _BITS_STDIO_H

#define EOF (-1)

int fprintf(FILE * stream, const char * fmt, ...);
int printf(const char * fmt, ...);
int snprintf(char *s, size_t n, const char * fmt, ...);
int sprintf(char *s, const char * fmt, ...);

#endif /* _BITS_STDIO_H */
