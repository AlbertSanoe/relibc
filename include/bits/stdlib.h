#ifndef _BITS_STDLIB_H
#define _BITS_STDLIB_H

# define abort() __abort(__func__, __FILE__, __LINE__)

#ifdef __cplusplus
extern "C" {
#endif

long double strtold(const char *nptr, char **endptr);

#ifdef __cplusplus
} // extern "C"
#endif

#endif /* _BITS_STDLIB_H */
