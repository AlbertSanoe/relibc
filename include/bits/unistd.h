#ifndef _BITS_UNISTD_H
#define _BITS_UNISTD_H

int execl(const char *path, const char* argv0, ...);
int execle(const char *path, const char* argv0, ...);

#endif
