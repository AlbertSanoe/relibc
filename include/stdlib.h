#ifndef _STDLIB_H
#define _STDLIB_H

#include <sys/types.h>

#endif /* _STDLIB_H */
