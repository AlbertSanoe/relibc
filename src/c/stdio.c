#include <stdarg.h>
#include <stddef.h>

typedef struct FILE FILE;

int vfprintf(FILE * stream, const char * fmt, va_list ap);

int fprintf(FILE * stream, const char * fmt, ...) {
    int ret;
    va_list ap;
    va_start(ap, fmt);
    ret = vfprintf(stream, fmt, ap);
    va_end(ap);
    return ret;
}

int vprintf(const char * fmt, va_list ap);

int printf(const char * fmt, ...) {
    int ret;
    va_list ap;
    va_start(ap, fmt);
    ret = vprintf(fmt, ap);
    va_end(ap);
    return ret;
}

int vsnprintf(char * s, size_t n, const char * fmt, va_list ap);

int snprintf(char * s, size_t n, const char * fmt, ...) {
    int ret;
    va_list ap;
    va_start(ap, fmt);
    ret = vsnprintf(s, n, fmt, ap);
    va_end(ap);
    return ret;
}

int vsprintf(char * s, const char * fmt, va_list ap);

int sprintf(char *s, const char * fmt, ...) {
    int ret;
    va_list ap;
    va_start(ap, fmt);
    ret = vsprintf(s, fmt, ap);
    va_end(ap);
    return ret;
}
