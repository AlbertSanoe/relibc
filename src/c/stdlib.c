double strtod(const char *nptr, char **endptr);

long double strtold(const char *nptr, char **endptr) {
    return (long double)strtod(nptr, endptr);
}

// manually define detailed abort function
void __abort(const char *func, const char *file, int line);

// backup definition of abort for programs that link it directly
void abort(void) {
    // call detailed abort function
    __abort(__func__, __FILE__, __LINE__);
}
