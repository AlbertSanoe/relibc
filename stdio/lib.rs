#[no_mangle]
pub extern "C" fn clearerr(arg1: *mut FILE) {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn ctermid(arg1: *mut libc::c_char)
     -> *mut libc::c_char {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn cuserid(arg1: *mut libc::c_char)
     -> *mut libc::c_char {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn fclose(arg1: *mut FILE) -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn fdopen(arg1: libc::c_int,
                  arg2: *const libc::c_char) -> *mut FILE {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn feof(arg1: *mut FILE) -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn ferror(arg1: *mut FILE) -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn fflush(arg1: *mut FILE) -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn fgetc(arg1: *mut FILE) -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn fgetpos(arg1: *mut FILE, arg2: *mut fpos_t)
     -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn fgets(arg1: *mut libc::c_char,
                 arg2: libc::c_int, arg3: *mut FILE)
     -> *mut libc::c_char {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn fileno(arg1: *mut FILE) -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn flockfile(arg1: *mut FILE) {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn fopen(arg1: *const libc::c_char,
                 arg2: *const libc::c_char) -> *mut FILE {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn fprintf(arg1: *mut FILE, arg2: *const libc::c_char, ...)
     -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn fputc(arg1: libc::c_int, arg2: *mut FILE)
     -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn fputs(arg1: *const libc::c_char, arg2: *mut FILE)
     -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn fread(arg1: *mut libc::c_void, arg2: usize, arg3: usize,
                 arg4: *mut FILE) -> usize {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn freopen(arg1: *const libc::c_char,
                   arg2: *const libc::c_char, arg3: *mut FILE)
     -> *mut FILE {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn fscanf(arg1: *mut FILE, arg2: *const libc::c_char, ...)
     -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn fseek(arg1: *mut FILE, arg2: libc::c_long,
                 arg3: libc::c_int) -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn fseeko(arg1: *mut FILE, arg2: off_t, arg3: libc::c_int)
     -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn fsetpos(arg1: *mut FILE, arg2: *const fpos_t)
     -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn ftell(arg1: *mut FILE) -> libc::c_long {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn ftello(arg1: *mut FILE) -> off_t {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn ftrylockfile(arg1: *mut FILE) -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn funlockfile(arg1: *mut FILE) {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn fwrite(arg1: *const libc::c_void, arg2: usize,
                  arg3: usize, arg4: *mut FILE) -> usize {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn getc(arg1: *mut FILE) -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn getchar() -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn getc_unlocked(arg1: *mut FILE) -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn getchar_unlocked() -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn getopt(arg1: libc::c_int,
                  arg2: *const *const libc::c_char,
                  arg3: libc::c_char) -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn gets(arg1: *mut libc::c_char)
     -> *mut libc::c_char {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn getw(arg1: *mut FILE) -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn pclose(arg1: *mut FILE) -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn perror(arg1: *const libc::c_char) {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn popen(arg1: *const libc::c_char,
                 arg2: *const libc::c_char) -> *mut FILE {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn printf(arg1: *const libc::c_char, ...)
     -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn putc(arg1: libc::c_int, arg2: *mut FILE)
     -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn putchar(arg1: libc::c_int) -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn putc_unlocked(arg1: libc::c_int, arg2: *mut FILE)
     -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn putchar_unlocked(arg1: libc::c_int)
     -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn puts(arg1: *const libc::c_char) -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn putw(arg1: libc::c_int, arg2: *mut FILE)
     -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn remove(arg1: *const libc::c_char)
     -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn rename(arg1: *const libc::c_char,
                  arg2: *const libc::c_char)
     -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn rewind(arg1: *mut FILE) {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn scanf(arg1: *const libc::c_char, ...)
     -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn setbuf(arg1: *mut FILE, arg2: *mut libc::c_char) {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn setvbuf(arg1: *mut FILE, arg2: *mut libc::c_char,
                   arg3: libc::c_int, arg4: usize)
     -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn snprintf(arg1: *mut libc::c_char, arg2: usize,
                    arg3: *const libc::c_char, ...)
     -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn sprintf(arg1: *mut libc::c_char,
                   arg2: *const libc::c_char, ...)
     -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn sscanf(arg1: *const libc::c_char,
                  arg2: *const libc::c_char, ...)
     -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn tempnam(arg1: *const libc::c_char,
                   arg2: *const libc::c_char)
     -> *mut libc::c_char {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn tmpfile() -> *mut FILE {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn tmpnam(arg1: *mut libc::c_char)
     -> *mut libc::c_char {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn ungetc(arg1: libc::c_int, arg2: *mut FILE)
     -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn vfprintf(arg1: *mut FILE, arg2: *const libc::c_char,
                    va_list: libc::c_int) -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn vprintf(arg1: *const libc::c_char,
                   va_list: libc::c_int) -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn vsnprintf(arg1: *mut libc::c_char, arg2: usize,
                     arg3: *const libc::c_char,
                     va_list: libc::c_int) -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn vsprintf(arg1: *mut libc::c_char,
                    arg2: *const libc::c_char,
                    va_list: libc::c_int) -> libc::c_int {
    unimplemented!();
}

