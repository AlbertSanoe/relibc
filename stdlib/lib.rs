/* automatically generated by rust-bindgen */

pub type wchar_t = libc::c_int;

#[repr(C)]
#[derive(Debug, Copy)]
pub struct div_t {
    pub quot: libc::c_int,
    pub rem: libc::c_int,
}

impl Clone for div_t {
    fn clone(&self) -> Self {
        *self
    }
}
#[repr(C)]
#[derive(Debug, Copy)]
pub struct ldiv_t {
    pub quot: libc::c_long,
    pub rem: libc::c_long,
}
impl Clone for ldiv_t {
    fn clone(&self) -> Self {
        *self
    }
}
#[no_mangle]
pub extern "C" fn a64l(s: *const char)
#[no_mangle]
pub extern "C" fn abort() {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn abs(i: libc::c_int) -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn atexit(func: ::std::option::Option<unsafe extern "C" fn()>) -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn atof(str_: *const libc::c_char) -> f64 {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn atoi(str_: *const libc::c_char) -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn atol(str_: *const libc::c_char) -> libc::c_long {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn bsearch(
    key: *const libc::c_void,
    base: *const libc::c_void,
    nel: usize,
    width: usize,
    compar: ::std::option::Option<
        unsafe extern "C" fn(arg1: *const libc::c_void,
                             arg2: *const libc::c_void)
                             -> libc::c_int,
    >,
) -> *mut libc::c_void {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn calloc(nelem: usize, elsize: usize) -> *mut libc::c_void {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn div(numer: libc::c_int, denom: libc::c_int) -> div_t {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn drand48() -> f64 {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn ecvt(
    value: f64,
    ndigit: libc::c_int,
    decept: *mut libc::c_int,
    sign: *mut libc::c_int,
) -> *mut libc::c_char {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn erand48(xsubi: *mut libc::c_ushort) -> f64 {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn exit(status: libc::c_int) {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn fcvt(
    value: f64,
    ndigit: libc::c_int,
    decept: *mut libc::c_int,
    sign: *mut libc::c_int,
) -> *mut libc::c_char {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn free(ptr: *mut libc::c_void) {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn gcvt(value: f64, ndigit: libc::c_int, buf: *mut libc::c_char) -> *mut libc::c_char {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn getenv(name: *const libc::c_char) -> *mut libc::c_char {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn getsubopt(
    optionp: *mut *mut libc::c_char,
    tokens: *const *const libc::c_char,
    valuep: *mut *mut libc::c_char,
) -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn grantpt(fildes: libc::c_int) -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn initstate(
    seed: libc::c_uint,
    state: *mut libc::c_char,
    size: usize,
) -> *mut libc::c_char {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn jrand48(xsubi: *mut libc::c_ushort) -> libc::c_long {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn l64a(value: libc::c_long) -> *mut libc::c_char {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn labs(i: libc::c_long) -> libc::c_long {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn lcong48(param: *mut libc::c_ushort) {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn ldiv(numer: libc::c_long, denom: libc::c_long) -> ldiv_t {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn lrand48() -> libc::c_long {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn malloc(size: usize) -> *mut libc::c_void {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn mblen(s: *const libc::c_char, n: usize) -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn mbstowcs(pwcs: *mut wchar_t, s: *const libc::c_char, n: usize) -> usize {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn mbtowc(
    pwc: *mut wchar_t,
    s: *const libc::c_char,
    n: usize,
) -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn mktemp(template: *mut libc::c_char) -> *mut libc::c_char {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn mkstemp(template: *mut libc::c_char) -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn mrand48() -> libc::c_long {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn nrand48(xsubi: *mut libc::c_ushort) -> libc::c_long {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn ptsname(fildes: libc::c_int) -> *mut libc::c_char {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn putenv(string: *mut libc::c_char) -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn qsort(
    base: *mut libc::c_void,
    nel: usize,
    width: usize,
    compar: ::std::option::Option<
        unsafe extern "C" fn(arg1: *const libc::c_void,
                             arg2: *const libc::c_void)
                             -> libc::c_int,
    >,
) {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn rand() -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn rand_r(seed: *mut libc::c_uint) -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn random() -> libc::c_long {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn realloc(ptr: *mut libc::c_void, size: usize) -> *mut libc::c_void {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn realpath(
    file_name: *const libc::c_char,
    resolved_name: *mut libc::c_char,
) -> *mut libc::c_char {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn seed48(seed16v: *mut libc::c_ushort) -> libc::c_ushort {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn setkey(key: *const libc::c_char) {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn setstate(state: *const libc::c_char) -> *mut libc::c_char {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn srand(seed: libc::c_uint) {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn srand48(seedval: libc::c_long) {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn srandom(seed: libc::c_uint) {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn strtod(str_: *const libc::c_char, endptr: *mut *mut libc::c_char) -> f64 {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn strtol(
    str_: *const libc::c_char,
    endptr: *mut *mut libc::c_char,
    base: libc::c_int,
) -> libc::c_long {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn strtoul(
    str_: *const libc::c_char,
    endptr: *mut *mut libc::c_char,
    base: libc::c_int,
) -> libc::c_ulong {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn system(command: *const libc::c_char) -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn ttyslot() -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn unlockpt(fildes: libc::c_int) -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn valloc(size: usize) -> *mut libc::c_void {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn wcstombs(s: *mut libc::c_char, pwcs: *const wchar_t, n: usize) -> usize {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn wctomb(s: *mut libc::c_char, wchar: wchar_t) -> libc::c_int {
    unimplemented!();
}
