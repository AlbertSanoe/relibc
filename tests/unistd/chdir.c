#include <limits.h>
#include <unistd.h>
#include <stdio.h>
#include <stdlib.h>

int main(void) {
    char cwd[PATH_MAX] = { 0 };
    char *cwd_result = NULL;

    cwd_result = getcwd(cwd, PATH_MAX);
    if (cwd_result == NULL) {
        perror("getcwd");
        exit(EXIT_FAILURE);
    } else if (cwd_result != cwd) {
        puts("getcwd returned something else than the buf argument");
        exit(EXIT_FAILURE);
    }

    printf("getcwd before chdir: %s\n", cwd);

    int status = chdir("..");
    if (status == -1) {
        perror("chdir");
        exit(EXIT_FAILURE);
    } else if (status != 0) {
        printf("chdir returned %d, unexpected result\n", status);
        exit(EXIT_FAILURE);
    }

    cwd_result = getcwd(cwd, PATH_MAX);
    if (cwd_result == NULL) {
        perror("getcwd");
        exit(EXIT_FAILURE);
    } else if (cwd_result != cwd) {
        puts("getcwd returned something else than the buf argument");
        exit(EXIT_FAILURE);
    }

    printf("getcwd after chdir: %s\n", cwd);
}
