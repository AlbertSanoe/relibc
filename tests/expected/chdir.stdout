initial cwd: /home/jeremy/Projects/relibc/tests
final cwd: /home/jeremy/Projects/relibc
